pyccel/codegen/printing/ccode.py

```diff
@@ -268,6 +268,7 @@ def __init__(self, parser, settings=None):
         self._parser = parser
         self._additional_code = ''
         self._additional_declare = []
+        self._unnecessary_declaration = []
```

Comment:
can you choose a meaningful name here. 
this dictionary is used to hold only declared variables in the `return` statement.

Reply:
No, it used to hold unnecessary declarations.

Reply:
we use it `_print_FunctionDef` to avoid declaring unused vars inside it.

Reply:
can't you just delete that variable from  `self._additional_declare`

Reply:
but its not in `self._additional_declare` other wise it will be declared in `_print_functiondef` in this line `decs += [Declare(i.dtype, i) for i in self._additional_declare]`, and i think this line is needed in other stuff.

Reply:
other than that i think i should store the unused variable in a normal variable instead of list since we assigned just a single scalar return or an expression to the unused variable.

Reply:
ah yes you are right

Reply:
how about deleting the variable from the Scope ?

Reply:
what do you mean by the Scope ?

Reply:
>how about deleting the variable from the Scope ?

These are not local variables. I don't think they are in the Scope. Instead they are simply temporary variables created from the results property of the class `FunctionDef`. The variables which should be printed could be selected and added to `self._additional_declare`, or we can assume that all are declared, and create a second set of variables for things that we thought we should declare but were deemed unnecessary later (in this case in `_print_Return`)

Reply:
they are in the Scope, because we need to infere their type in the semantic stage.

Reply:
You are right, it is done [here](https://github.com/pyccel/pyccel/blob/master/pyccel/parser/semantic.py#L2545). I thought these were created in the semantic stage, sorry. Why do we need to delete them from the Scope at this point?

Reply:
it's the easiest thing no ? instead of ingnoring them, we can just delete them so that we don't declare them .

Reply:
We don't declare them from the scope. They are declared [here](https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/ccode.py#L956-L957)